### ISSUE TITLE
Strange behavior when importing `from` TF submodules

### ISSUE THREAD

**Issue by jvmancuso** [tensorflow/tensorflow#33654]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes, but have tested in a standalone bare Python script
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): OS X 10.14
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.0.0
- Python version: 3.5
- CUDA/cuDNN version: None
- GPU model and memory: None

**Describe the current behavior**
```python
from tensorflow import data  # ==> imports the tf.data submodule
from tensorflow.data import Dataset  # ==> Raises: ImportError: No module named 'tensorflow.data'
```

**Describe the expected behavior**
I would expect the Dataset class to be imported.

**Code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate the problem.
```python
from tensorflow.data import Dataset
```

**Comment by oanush** [comment]
@jvmancuso ,
Hello, yes for TF 2.0 version please try using 
`from tensorflow import 
 data data=data.dataset` it should work fine.
Find [gist](https://colab.sandbox.google.com/gist/oanush/a21d362af3a3c38024364fea7dab3237/33654.ipynb) colab  for the same.Thanks!

**Comment by jvmancuso** [comment]
Yes, that works!  However, that seems to be a workaround/anti-pattern and not a fix, so still seems like a bug to me.

**Comment by mihaimaruseac** [comment]
See https://github.com/tensorflow/tensorflow/issues/32957#issuecomment-543819065

TL;DR: the only supported API usage is

```
import tensorflow as tf
tf.get.here.all.the.attributes.you.want
```

Any other API usage can break at any time and fixing it is a best effort job.

(I linked to that comment as it proves that if you can do `from module import name` and `name.other_name` that doesn't imply you can also do `from module.name import other_name` always)
